### PR TITLE
Honor the Disable Visual Editor setting

### LINKED ISSUE #4634: Should Gutenberg respect the rich_editing user option?
In the classic editor, a WordPress user can disable the `rich_editing` option. If this is disabled, then TinyMCE isn't loaded when editing a post.

Should Gutenberg also respect the `rich_editing` user option, or does this feature become deprecated?

From https://wordpress.slack.com/archives/C02QB2JS7/p1516553718000137

### PR DESCRIPTION
**Pull request by mkaz**
## Description

Closes #4634.

Show the plain text editor if the user checks the "Disable the visual editor when writing" option
See #4634 

## How has this been tested?

From user settings, check option to Disable the visual editor:

* Edit post from Post list open in plain text editor
* Edit post from link on page view opens in plain text
* Switching back to disable option brings back block editor
* Confirm paragraphs, embeds, and other content works as expected

## Types of changes

* Removes the forcing of user_can_richedit to false
* Disables Gutenberg init by returning if user_can_richedit()
* Removes Gutenberg admin menu item
* Removes Classic Editor link in post view

## Checklist:
- [X] My code is tested.
- [X] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation.

### CODE REVIEW

**Review by youknowriad — APPROVED** [review]
LGTM 👍

**Comment by danielbachhuber** [comment]
@gziolo @youknowriad Is there a core Trac ticket for this?

**Comment by youknowriad** [comment]
@danielbachhuber Not that I'm aware of? Is the process to create a trac ticket for each php change? Or do we do git diffs since last updates or something? Just trying to understand the process here?

**Comment by danielbachhuber** [comment]
> Is the process to create a trac ticket for each php change?

Yes.

File: gutenberg.php [hunk 1]
```diff
@@ -67,6 +67,9 @@ function the_gutenberg_project() {
  * @since 0.1.0
  */
 function gutenberg_menu() {
+	if ( ! gutenberg_can_edit_post( $post ) ) {
+		return;
+	}
```
Comment (youknowriad) [inline gutenberg.php]: This check is creating a warning and for me it doesn't make sense to check this here at all. It's being addressed in #12053

**Comment by youknowriad** [comment]
Trac ticket https://core.trac.wordpress.org/ticket/45375

**Pull request merged**